BDArmory/Guidances/MissileGuidance.cs

```diff
@@ -243,7 +252,24 @@ public static Vector3 GetWeaveTarget(Vector3 targetPosition, Vector3 targetVeloc
 
             float leadTime = Mathf.Min(4f, ttgoWeave);
 
-            return missileVessel.CoM + leadTime * missileVel + accel * (0.5f * leadTime * leadTime);
+            Vector3 aimPos = missileVessel.CoM + leadTime * missileVel + accel * (0.5f * leadTime * leadTime);
+
+            if (useAGMDescentRatio)
+            {
+                float altitudeClamp = Mathf.Clamp(
+                    (weaveDist - ((float)missileVessel.srfSpeed * agmDescentRatio)) * 0.22f, 0f,
+                    (float)missileVessel.altitude +
+                    Mathf.Max(VectorUtils.AnglePreNormalized(missileVessel.upAxis, missileVel.normalized) - 90f, 0f) * Mathf.Deg2Rad * aimPos.magnitude);
```

Comment:
`aimPos` is an absolute position, which depends on the floating origin, so I don't think this is what you intended.
I suspect you meant `(aimPos - missileVessel.COM).magnitude`.

Also, for consistency, you could use `upDirection` instead of `missileVessel.upAxis` here.